config/ola.m4

```diff
@@ -110,19 +132,36 @@ AC_DEFUN([PLUGIN_SUPPORT],
 
   AC_ARG_ENABLE(
     [$1],
-    AS_HELP_STRING([--disable-$1], [Disable the $1 plugin]))
+    AS_HELP_STRING([--disable-$1], [Disable the $1 plugin]),
+    [],
+    [eval $enable_arg=auto])
 
   eval enable_plugin=\$$enable_arg;
+
+  # Input validation.
+  if test "${enable_plugin}" != "yes" -a \
+          "${enable_plugin}" != "no" -a \
+          "${enable_plugin}" != "auto"; then
+    AC_MSG_ERROR([Invalid value for --enable-${plugin_key}/\
+--disable-${plugin_key}. Valid values are yes, no and auto.])
+  fi
+
+  # If dependencies are not met...
   if test "$3" == "no"; then
+    # ...and the user has explicitely requested this plugin to be enabled,
+    # error out.
+    if test "${enable_plugin}" = "yes"; then
+      AC_MSG_ERROR([Dependencies for plugin ${plugin_key} are not met.])
```

Comment:
Or a multi-dimensional array (which I guess is roughly the same). Probably one for a seperate PR anyway.